### PR TITLE
Migrate ResourceLoader to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate ResourceLoader to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/ResourceLoader.kt [hunk 1]
```diff
@@ -14,76 +14,67 @@
...
-                throw new FileNotFoundException("Could not find test file: " + name);
-            }
-            File file = new File(getTestDir(context, name), newName);
-            String dst = file.getAbsolutePath();
-            Utils.writeToFile(is, dst);
-            file.deleteOnExit();
-            return dst;
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+    @KotlinCleanup("is --> equalTo")
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ResourceLoader.kt:34]: `inputStream`

File: AnkiDroid/src/test/java/com/ichi2/utils/ResourceLoader.kt [hunk 2]
```diff
@@ -14,76 +14,67 @@
...
-     *      emptied on every invocation of this method so it is suitable to use at the start of each test.
-     *      Only add files (and not subdirectories) to this directory.
+     * emptied on every invocation of this method so it is suitable to use at the start of each test.
+     * Only add files (and not subdirectories) to this directory.
      */
-    private static File getTestDir(Context context, String name) {
-        String suffix = "";
+    private fun getTestDir(context: Context, name: String): File {
+        var suffix = ""
         if (!TextUtils.isEmpty(name)) {
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ResourceLoader.kt:62]: `@KotlinCleanup("name.isNotEmpty()")`

File: AnkiDroid/src/test/java/com/ichi2/utils/ResourceLoader.kt [hunk 3]
```diff
@@ -14,76 +14,67 @@
...
-            suffix = "-" + name;
+            suffix = "-$name"
         }
-        File dir = new File(context.getCacheDir(), "testfiles" + suffix);
+        val dir = File(context.cacheDir, "testfiles$suffix")
         if (!dir.exists()) {
-            assertTrue(dir.mkdir());
+            Assert.assertTrue(dir.mkdir())
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ResourceLoader.kt:67]: import `Assert`

**Review by david-allison — CHANGES_REQUESTED** [review]
looks good! only nitpicks

File: AnkiDroid/src/test/java/com/ichi2/utils/ResourceLoader.kt [hunk 4]
```diff
@@ -14,76 +14,69 @@
...
+object ResourceLoader {
     /**
      * Copy a file from the application's assets directory and return the absolute path of that
      * copy.
      *
      * Files located inside the application's assets collection are not stored on the file
      * system and can not return a usable path, so copying them to disk is a requirement.
      */
-    public static String getTempFilePath(Context context, String name, String newName) {
+    @KotlinCleanup("is --> equalTo")
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ResourceLoader.kt:35]:
```
    @KotlinCleanup("is --> inputStream")
```

**Review by david-allison — CHANGES_REQUESTED** [review]
one nit

**Review by david-allison — APPROVED** [review]
that was fast! Thank you!

**Review by mikehardy — APPROVED** [review]
excellent

**Pull request merged**